File: 2022/day1.2.js
function solvePuzzle(input) {
	const rows = input.split('\n')
	const topList = []
	let elfKcal = 0
	
	for (const row of rows) {
		const kcal = Number(row)

		if (kcal) {
			elfKcal += kcal
		} else {
			topList.push(elfKcal)
			elfKcal = 0

			if (topList.length > 3) {
				const smallestTotal = Math.min(...topList)
				const indexOfSmallestTotal = topList.indexOf(smallestTotal)
				topList.splice(indexOfSmallestTotal, 1)
			}
		}
	}

	return topList.reduce((acc, total) => acc + total, 0)
}

const input = document.querySelector('pre').innerText
const result = solvePuzzle(input)

console.log(result)

File: 2022/day1.2.py
with open("2022/input.txt", "r") as file:
    input = file.read()

MAX_TOP_ITEMS = 3


def solve_puzzle(input):
    kcal_top_list = []

    for elf_foods in input.split("\n\n"):
        kcals = map(int, elf_foods.split("\n"))
        elf_total = sum(kcals)
        kcal_top_list.append(elf_total)

        if len(kcal_top_list) > MAX_TOP_ITEMS:
            lowest = min(kcal_top_list)
            kcal_top_list.remove(lowest)

    return sum(kcal_top_list)


result = solve_puzzle(input)
print(result)
